tests/test_deltaplot_control_callback_safety.py

```
@@ -0,0 +1,252 @@
+        delta_plot(_DataSentinel(), _DataSentinel(), const_range=value)
+
+    assert value.calls == 0
+
+
+@pytest.mark.parametrize(
+    ("kwargs", "message"),
+    (
+        ({"threshold": "other"}, "threshold must be 'norm' or 'fixed'"),
+        ({"extreme": "other"}, "extreme must be 'constraint' or 'add'"),
+        ({"purify": "other"}, "purify must be None, 'IPP1', 'IPP2', or 'IPP3'"),
+        ({"alpha": 0.0}, "alpha must be finite and in \(0, 1\)"),
```

Ruff (0.16.1)

[warning] Invalid escape sequence: `\(`

Use a raw string literal

(W605)